File: docs/breez-sdk/snippets/kotlin_mpp_lib/shared/src/commonMain/kotlin/com/example/kotlinmpplib/LnurlAuth.kt
package com.example.kotlinmpplib

import breez_sdk_spark.*

suspend fun parseLnurlAuth(client: BreezClient) {
    // ANCHOR: parse-lnurl-auth
    // LNURL-auth URL from a service
    // Can be in the form:
    // - lnurl1... (bech32 encoded)
    // - https://service.com/lnurl-auth?tag=login&k1=...
    val lnurlAuthUrl = "lnurl1..."

    when (val inputType = client.parse(lnurlAuthUrl)) {
        is InputType.LnurlAuth -> {
            val requestData = inputType.v1
            println("Domain: ${requestData.domain}")
            println("Action: ${requestData.action}")

            // Show domain to user and ask for confirmation
            // This is important for security
        }
        else -> {}
    }
    // ANCHOR_END: parse-lnurl-auth
}

suspend fun authenticate(client: BreezClient, requestData: LnurlAuthRequestDetails) {
    // ANCHOR: lnurl-auth
    // Perform LNURL authentication
    when (val result = client.lnurl().auth(requestData)) {
        is LnurlCallbackStatus.Ok -> {
            println("Authentication successful")
        }
        is LnurlCallbackStatus.ErrorStatus -> {
            println("Authentication failed: ${result.errorDetails.reason}")
        }
    }
    // ANCHOR_END: lnurl-auth
}

File: docs/breez-sdk/snippets/swift/BreezSdkSnippets/Sources/LnurlAuth.swift
import BreezSdkSpark
import Foundation

func parseLnurlAuth(client: BreezClient) async throws {
    // ANCHOR: parse-lnurl-auth
    // LNURL-auth URL from a service
    // Can be in the form:
    // - lnurl1... (bech32 encoded)
    // - https://service.com/lnurl-auth?tag=login&k1=...
    let lnurlAuthUrl = "lnurl1..."

    if case .lnurlAuth(v1: let requestData) = try await client.parse(input: lnurlAuthUrl) {
        print("Domain: \(requestData.domain)")
        print("Action: \(String(describing: requestData.action))")

        // Show domain to user and ask for confirmation
        // This is important for security
    }
    // ANCHOR_END: parse-lnurl-auth
}

func authenticate(client: BreezClient, requestData: LnurlAuthRequestDetails) async throws {
    // ANCHOR: lnurl-auth
    // Perform LNURL authentication
    let result = try await client.lnurl().auth(requestData: requestData)

    switch result {
    case .ok:
        print("Authentication successful")
    case .errorStatus(errorDetails: let errorDetails):
        print("Authentication failed: \(errorDetails.reason)")
    }
    // ANCHOR_END: lnurl-auth
}
